Casks/u/ubiquiti-unifi-controller.rb

```diff
@@ -10,7 +10,16 @@
 
   livecheck do
     url "https://fw-update.ubnt.com/api/firmware-latest?filter=eq~~product~~unifi-controller&filter=eq~~channel~~release&filter=eq~~platform~~macos"
-    regex(/"version"\s*:\s*"v?(\d+(?:\.\d+)+)/i)
+    strategy :json do |json|
+      json.dig("_embedded", "firmware")&.map do |item|
+        major = item["version_major"]
+        minor = item["version_minor"]
+        patch = item["version_patch"]
+        next if major.blank? || minor.blank? || patch.blank?
+
+        "#{major}.#{minor}.#{patch}"
```

Comment:
```suggestion
    regex(/^\D*?(\d+(?:\.\d+)+)/i)
    strategy :json do |json, regex|
      json.dig("_embedded", "firmware")&.filter_map do |item|
        item["version"]&.[](regex, 1)
```

There may be something to be said for matching the version from the start of the `version` text (e.g., `v8.5.6+atag-8.5.6-27036`), as this aligns with the version in the `href` URL and would continue to work if upstream happens to release a version that doesn't include all of major/minor/patch. [We could adapt the `strategy` block logic to use something like `item["version_minor"] || 0` but we could end up with a broken `url` if upstream only uses something like `1.2` for a version instead of `1.2.0`.]

Upstream consistently uses major/minor/patch for all the visible releases but I try to avoid relying on the assumption that what's true for their version format now will always remain true in the future (if we can reasonably introduce some flexibility without creating issues).

Reply:
Makes sense! This one did feel a bit forced, after all.